wgengine/monitor/monitor_windows.go

```diff
@@ -136,47 +138,52 @@ func (m *winMon) getIPOrRouteChangeMessage() (message, error) {
 		return nil, errClosed
 	}
 
-	var o windows.Overlapped
-	h, err := windows.CreateEvent(nil, 1 /* true*/, 0 /* unsignaled */, nil /* no name */)
-	if err != nil {
-		m.logf("CreateEvent: %v", err)
-		return nil, err
-	}
-	defer windows.CloseHandle(h)
+	oaddr := new(windows.Overlapped)
+	oroute := new(windows.Overlapped)
 
-	m.mu.Lock()
-	m.event = h
-	m.mu.Unlock()
-
-	o.HEvent = h
-
-	err = notifyAddrChange(&h, &o)
+	err := notifyAddrChange(&oaddr.HEvent, oaddr)
 	if err != nil {
 		m.logf("notifyAddrChange: %v", err)
 		return nil, err
 	}
-	err = notifyRouteChange(&h, &o)
+	defer cancelIPChangeNotifyProc.Call(uintptr(unsafe.Pointer(oaddr)))
```

Comment:
This makes me a tiny bit nervous, because there could be a tiny window with nothing keeping oaddr alive. (We do the conversion at first evaluation, but the call on function exit.) Let's indulge in paranoia and keep the conversion close to use:

```go
defer func() {
    cancelIPChangeNotifyProc.Call(uintptr(unsafe.Pointer(oaddr)))
}()
```